### PR TITLE
Do not create issue when unable to determine if issue already exists

### PR DESCRIPTION
**Pull request by IEvangelist**
- Added a check to ensure that the HTTP call to GraphQL was successful
  - Call `response.EnsureSuccessStatusCode()`, this will `throw`
- Update the logic to conditionally create an issue, when, and only when we're able to verify that a dup doesn't already exist
- Minor unit test updates

Fixes: #36

### CODE REVIEW

File: src/DotNet.VersionSweeper/Program.cs [hunk 1]
```diff
@@ -52,9 +52,13 @@ static async Task StartSweeperAsync(Options options, IServiceProvider services,
             var existingIssue =
                 await client.GetIssueAsync(
                     options.Owner, options.Name, options.Token, title);
-            if (existingIssue?.State == ItemState.Open)
+            if (existingIssue is null)
```
Comment (BillWagner) [inline src/DotNet.VersionSweeper/Program.cs:55]: This set of logic looks backward to me. 

I'm reading it as this first `if` is true when there isn't an existing issue. On line 59, you check if there is an existing issue and it's open (indicating you shouldn't make a new issue). The final `else` would seem to be an issue exists, and it's closed. That should imply that it's fixed, right?

What am I missing?

**Review by BillWagner — COMMENTED** [review]
I like the idea behind this change. I wonder if I'm reading the logic incorrectly. Can you check my comments?

File: src/DotNet.VersionSweeper/Program.cs (hunk 1 above)
Comment (IEvangelist) [reply src/DotNet.VersionSweeper/Program.cs:55]: The way you're reading it it correct, in the first `if` when there isn't an existing issue - log and continue. The second condition is when the `else if` branch determines that there is an existing issue and it is still open, log and continue. Finally, if there are no existing open issues, create a new one. That is the intent, does that make sense?

**Review by BillWagner — APPROVED** [review]
Thanks @IEvangelist 

This LGTM. You can :shipit: when ready.

**Pull request merged**